src/event/player/PlayerViewDistanceChangeEvent.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\event\player;
+
+use pocketmine\event\Cancellable;
+use pocketmine\event\CancellableTrait;
+use pocketmine\player\Player;
+
+/**
+ * Called when a player requests a different viewing distance than the current one.
+ */
+class PlayerViewDistanceChangeEvent extends PlayerEvent implements Cancellable{
+	use CancellableTrait;
```

Comment:
I'm doubtful that this should be cancellable. It will cause bugs on the client unless synced back.

Reply:
Indeed, when the player connects, if a plugin cancels the event, it will be blocked

Reply:
What if the player changes the view distance while in-game?

Reply:
If the player is already logged in, this was not going to take their request. However, making this event cancellable is not the right solution because it can block some behavior.